### PR TITLE
[dbus] add OTBR_PACKAGE_VERSION

### PR DESCRIPTION
**Pull request by jdswensen**
Fetch the `otbr` package version via `dbus`

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

File: src/dbus/server/dbus_thread_object.cpp [hunk 1]
```diff
@@ -1499,6 +1501,17 @@ void DBusThreadObject::RegisterGetPropertyHandler(const std::string         &aIn
     mGetPropertyHandlers[aPropertyName] = aHandler;
 }
 
+otError DBusThreadObject::GetOtbrVersionHandler(DBusMessageIter &aIter)
+{
+    otError     error   = OT_ERROR_NONE;
+    std::string version = OTBR_PACKAGE_VERSION;
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, version) == OTBR_ERROR_NONE, error = OT_ERROR_FAILED);
```
Comment (superwhd) [inline src/dbus/server/dbus_thread_object.cpp:1509]: nit: I think this can be simplified by `SuccessOrExit`

**Review by superwhd — APPROVED** [review]
LGTM with a nit

File: src/dbus/server/dbus_thread_object.cpp (hunk 1 above)
Comment (jdswensen) [reply src/dbus/server/dbus_thread_object.cpp]: Change made!

Are there situations in which `SuccessOrExit` is preferred over `VerifyOrExit`? Or is there a push to move towards using `SuccessOrExit` in general when `OTBR_ERROR_NONE` is expected?

Comment (superwhd) [reply src/dbus/server/dbus_thread_object.cpp]: I think `SuccessOrExit` is preferred when the checked value is an `OTBR_ERROR_*`.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**